src/ChainSafe.GamingSdk.Gelato/Gelato.cs

```diff
@@ -0,0 +1,262 @@
+                callRequest.UserNonce ??= optional.UserNonce;
+                return await gelatoClient.Post<CallWithErc2771Request, RelayResponse>(RelayCall.SponsoredCallErc2771, callRequest);
+            }
+            catch (Exception e)
+            {
+                throw new Web3Exception($"GelatoRelaySDK/sponsoredCallErc2771: Failed with error: {e.Message}");
+            }
+        }
+
+        public async Task<HexBigInteger> GetEstimatedFee(string paymentToken, HexBigInteger gasLimit, bool isHighPriority, HexBigInteger gasLimitL1 = null)
+        {
+            if (gasLimitL1 == null)
+            {
+                gasLimitL1 = new HexBigInteger("0x0");
+            }
+
+            var request = new EstimatedFeeRequest
+            {
+                ChainId = chainConfig.ChainId,
+                GasLimit = gasLimit,
+                GasLimitL1 = gasLimitL1,
+                IsHighPriority = isHighPriority,
+                PaymentToken = paymentToken,
+            };
+
+            return await gelatoClient.GetEstimatedFeeRequest(request);
+        }
+
+        private async Task<bool> IsNetworkSupported(string networkId)
+        {
+            var supportedNetworks = await gelatoClient.GetSupportedNetworks();
+            return supportedNetworks.Contains(networkId);
+        }
+
+        private async Task<bool> IsOracleActive(string chainId)
+        {
+            var oracles = await gelatoClient.GetGelatoOracles();
+            return oracles.Contains(chainId);
+        }
+
+        private SerializableDomain GetEip712Domain(Erc2771Type type)
+        {
+            return type switch
+            {
+                Erc2771Type.CallWithSyncFee => new SerializableDomain()
+                {
+                    Name = "GelatoRelayERC2771",
+                    Version = "1",
+                    ChainId = chainConfig.ChainId,
+                    VerifyingContract = CallWithErc2771RequestOptionalParameters
+                        .GetGelatoRelayErc2771Address(type, config, chainConfig),
+                },
+                Erc2771Type.SponsoredCall => new SerializableDomain()
+                {
+                    Name = "GelatoRelay1BalanceERC2771",
+                    Version = "1",
+                    ChainId = chainConfig.ChainId,
+                    VerifyingContract = CallWithErc2771RequestOptionalParameters
+                        .GetGelatoRelayErc2771Address(type, config, chainConfig),
+                },
+                _ => throw new Web3Exception("incorrect relay option")
+            };
+        }
+
+        public async Task<RelayedTask> GetTaskStatus(string taskId)
+        {
+            return await gelatoClient.GetTaskStatus(taskId);
+        }
+
+        public async Task<string[]> GetPaymentTokens()
+        {
+            return await gelatoClient.GetPaymentTokens(chainConfig.ChainId);
+        }
+
+        public async ValueTask WillStartAsync()
```

Comment:
It makes more sense to put initialization/deinitialization code right after the constructors' declaration so that all the logic related to initialization/deinitialization is grouped at the top.